File: src/main/java/fr/univamu/iut/commandes/Commandes.java
package fr.univamu.iut.commandes;

/**
 * Classe représentant une commande
 */
public class Commandes {

    protected int id_commande;
    protected double prix_total;
    protected String date_retrait;
    protected String localisation_retrait;
    protected String statut;

    public Commandes() {}

    public Commandes(int id_commande, double prix_total, String date_retrait, String localisation_retrait, String statut) {
        this.id_commande = id_commande;
        this.prix_total = prix_total;
        this.date_retrait = date_retrait;
        this.localisation_retrait = localisation_retrait;
        this.statut = statut;
    }

    // Getters et Setters pour chaque attribut
    public int getId_commande() {
        return id_commande;
    }

    public void setId_commande(int id_commande) {
        this.id_commande = id_commande;
    }

    public double getPrix_total() {
        return prix_total;
    }

    public void setPrix_total(double prix_total) {
        this.prix_total = prix_total;
    }

    public String getDate_retrait() {
        return date_retrait;
    }

    public void setDate_retrait(String date_retrait) {
        this.date_retrait = date_retrait;
    }

    public String getLocalisation_retrait() {
        return localisation_retrait;
    }

    public void setLocalisation_retrait(String localisation_retrait) {
        this.localisation_retrait = localisation_retrait;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    @Override
    public String toString() {
        return "Commande{" +
                "id_commande=" + id_commande +
                ", prix_total=" + prix_total +
                ", date_retrait='" + date_retrait + '\'' +
                ", localisation_retrait='" + localisation_retrait + '\'' +
                ", statut='" + statut + '\'' +
                '}';
    }
}

File: domain/commandes.php
<?php

namespace domain;
class commandes
{
    protected $id_commande;
    protected $id_panier;
    protected $id_abonne;
    protected $moyen_paiement;
    protected $localisation_retrait;
    protected $date_commande;
    protected $date_retrait;
    protected $prix_total;
    protected $statues;


    public function __construct($id_commande, $id_panier, $id_abonne, $moyen_paiement, $localisation_retrait, $date_commande, $date_retrait, $prix_total, $statues)
    {
        $this->id_commande = $id_commande;
        $this->id_panier = $id_panier;
        $this->id_abonne = $id_abonne;
        $this->moyen_paiement = $moyen_paiement;
        $this->localisation_retrait = $localisation_retrait;
        $this->date_commande = $date_commande;
        $this->date_retrait = $date_retrait;
        $this->prix_total = $prix_total;
        $this->statues = $statues;
    }


    public function getIdCommande()
    {
        return $this->id_commande;
    }
    public function getIdPanier()
    {
        return $this->id_panier;
    }
    public function getIdAbonne()
    {
        return $this->id_abonne;
    }
    public function getMoyenPaiement()
    {
        return $this->moyen_paiement;
    }
    public function getLocalisationRetrait()
    {
        return $this->localisation_retrait;
    }
    public function getDateCommande()
    {
        return $this->date_commande;
    }
    public function getDateRetrait()
    {
        return $this->date_retrait;
    }
    public function getPrixTotal()
    {
        return $this->prix_total;
    }
    public function getStatues()
    {
        return $this->statues;
    }
}
